crm_compassion/models/weekly_demand.py

```diff
@@ -99,61 +113,104 @@ def _compute_demand_events(self):
+            ('start_date', '>=', fields.Date.to_string(start_date))
+        ])
+        return float(sub_sponsored) / STATS_DURATION
+
     @api.depends('number_children_website', 'number_children_ambassador',
-                 'number_children_events')
+                 'number_sub_sponsorship', 'number_children_events')
     @api.multi
     def _compute_demand_total(self):
         for week in self:
             week.total_demand = week.number_children_website + \
+                week.number_sub_sponsorship + \
                 week.number_children_ambassador + week.number_children_events
 
     @api.model
     def _default_unsponsored_web(self):
         """ Compute average of unsponsored children on the website since
         one year.
         """
-        start_date = datetime.today() - timedelta(weeks=52)
+        start_date = datetime.today() - timedelta(weeks=STATS_DURATION)
         web_sponsored = self.env['recurring.contract'].search_count([
             ('channel', '=', 'internet'),
             ('start_date', '>=', fields.Date.to_string(start_date))
         ])
         allocate_per_week = int(self.env['ir.config_parameter'].get_param(
             'crm_compassion.number_children_web'))
-        return allocate_per_week - (float(web_sponsored) / 52.0)
+        return allocate_per_week - (float(web_sponsored) / STATS_DURATION)
 
     @api.model
     def _default_unsponsored_ambassador(self):
         """ Compute average of unsponsored children from ambassadors since
         one year.
         """
-        start_date = datetime.today() - timedelta(weeks=52)
+        start_date = datetime.today() - timedelta(weeks=STATS_DURATION)
         ambass_sponsored = self.env['recurring.contract'].search_count([
             ('origin_id.type', '=', 'partner'),
             ('origin_id.partner_id', '!=', False),
             ('start_date', '>=', fields.Date.to_string(start_date))
         ])
         allocate_per_week = int(self.env['ir.config_parameter'].get_param(
             'crm_compassion.number_children_ambassador'))
-        return allocate_per_week - (float(ambass_sponsored) / 52.0)
+        return allocate_per_week - (float(ambass_sponsored) / STATS_DURATION)
+
+    @api.depends('week_start_date')
+    @api.multi
+    def _compute_resupply_sub(self):
+        """ Compute SUB resupply. """
+        sub_average = self._default_demand_sub()
+        for week in self:
+            today = datetime.today()
+            start_date = fields.Datetime.from_string(
+                week.week_start_date) - timedelta(days=SUB_DURATION)
+            if start_date <= today:
+                rejected_sub = self.env['recurring.contract'].search_count([
+                    ('origin_id.type', '=', 'sub'),
+                    ('start_date', '>=', fields.Date.from_string(
+                        week.week_start_date) - timedelta(days=SUB_DURATION))
+                ])
+                sub_reject_average = rejected_sub / STATS_DURATION
```

Comment:
This is not the sub_reject_average. Here you do the sub average per day since 90 days.